fairseq/models/transformer.py

```diff
@@ -280,13 +291,24 @@ def reorder_encoder_out(self, encoder_out, new_order):
         if encoder_out['encoder_padding_mask'] is not None:
             encoder_out['encoder_padding_mask'] = \
                 encoder_out['encoder_padding_mask'].index_select(0, new_order)
+        if encoder_out['encoder_states'] is not None:
+            for idx, state in enumerate(encoder_out['encoder_states']):
+                encoder_out['encoder_states'][idx] = state.index_select(1, new_order)
         return encoder_out
 
     def max_positions(self):
         """Maximum input length supported by the encoder."""
         if self.embed_positions is None:
             return self.max_source_positions
         return min(self.max_source_positions, self.embed_positions.max_positions())
+        
+    def buffered_future_mask(self, tensor):
```

Comment:
Is this actually needed in the encoder? I can't see where it's used. There's already a definition in the decoder if that's sufficient?